client/app/dedicated/server/dedicated-server.service.js

```diff
@@ -1328,6 +1328,101 @@ angular
       }).then(details => details, () => null);
     };
 
+    this.getBareMetalPrivateBandwidthOptions = function (productId, existingBandwidth) {
+      return OvhHttp.get('/order/upgrade/baremetalPrivateBandwidth/{serviceName}', {
+        rootPath: 'apiv6',
+        urlParams: {
+          serviceName: productId,
+        },
+      }).then((planList) => {
+        const list = _.compact(_.map(planList, (plan) => {
```

Comment:
Please extract into multiple functions or add comments, this is hard to read :)

Reply:
And you should `return` directly.